parquet-cli/src/main/java/org/apache/parquet/cli/Util.java

```diff
@@ -96,34 +91,14 @@ public static String minMaxAsString(Statistics stats, OriginalType annotation) {
       return "";
     }
     // TODO: use original types when showing decimal, timestamp, etc.
-    if (stats instanceof BooleanStatistics) {
-      return String.format("%s / %s",
-          ((BooleanStatistics) stats).getMin(),
-          ((BooleanStatistics) stats).getMax());
-    } else if (stats instanceof IntStatistics) {
-      return String.format("%d / %d",
-          ((IntStatistics) stats).getMin(),
-          ((IntStatistics) stats).getMax());
-    } else if (stats instanceof LongStatistics) {
-      return String.format("%d / %d",
-          ((LongStatistics) stats).getMin(),
-          ((LongStatistics) stats).getMax());
-    } else if (stats instanceof FloatStatistics) {
-      return String.format("%f / %f",
-          ((FloatStatistics) stats).getMin(),
-          ((FloatStatistics) stats).getMax());
-    } else if (stats instanceof DoubleStatistics) {
-      return String.format("%f / %f",
-          ((DoubleStatistics) stats).getMin(),
-          ((DoubleStatistics) stats).getMax());
-    } else if (stats instanceof BinaryStatistics) {
+    if (stats instanceof BinaryStatistics) {
```

Comment:
This is where the other stats types have been removed. Why are only binary statistics supported here?

Reply:
Now, I got it. There was another concept I've started with and I've forgot to revert this change. I'll do it in the next commit.
Thanks for the finding.

Reply:
I've designed the new `PrimitiveComparator` super class to have primitive comparisons to avoid the unnecessary boxing of primitive types. So, I would keep the current specialised `Statistics` classes for performance reasons.

Reply:
+1